---
title: "diffbind_step5_overlaps"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Do differentially bound genes overlap differentially knockdown genes?

Read annotated peaks
```{r}
peaks_7v1_anno <- readRDS(file = "../data_output/diffbind_cutnrun_zic/peaks_7v1_up_annotation.Rds")
peaks_1v7_anno <- readRDS( file = "../data_output/diffbind_cutnrun_zic/peaks_7v1_down_annotation.Rds")
peaks_all_anno <- readRDS(file = "../data_output/diffbind_cutnrun_zic/peaks_7v1_all_annotation.Rds")

```

```{r}
peaks_anno_list <- list("D7" = peaks_7v1_anno,
                 "D1" = peaks_1v7_anno,
                 "all" = peaks_all_anno)

gene_anno_list <- lapply(peaks_anno_list,
       function(x) x@anno[x@detailGenomicAnnotation$Promoter])

```


```{r}

genes_d7 <- unique(gene_anno_list$D7$SYMBOL )
genes_d1 <- unique(gene_anno_list$D1$SYMBOL)
genes_all <- unique(gene_anno_list$all$SYMBOL)
```


Read differentially expressed genes

```{r}
knockdown_genes_zic1 <- readxl::read_excel("../data_input/melyssa_zic_data/41593_2015_BFnn3995_MOESM88_ESM.xls")
knockdown_genes_zic2 <- readxl::read_excel("../data_input/melyssa_zic_data/41593_2015_BFnn3995_MOESM88_ESM.xls",sheet = 2)

```

```{r}
knockdown_genes_zic1_down <- knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)` < 0]
knockdown_genes_zic1_up <- knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)` > 0]
knockdown_genes_zic2_down <- knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)` < 0]
knockdown_genes_zic2_up <- knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)` > 0]
```


```{r}
sum(genes_d1 %in% knockdown_genes_zic1$gene)
sum(genes_d7 %in% knockdown_genes_zic1$gene)

sum(genes_d1 %in% knockdown_genes_zic2$gene)
sum(genes_d7 %in% knockdown_genes_zic2$gene)

genes_d1[genes_d1 %in% knockdown_genes_zic1$gene]

genes_d1[genes_d1 %in% knockdown_genes_zic2$gene]

genes_d7[genes_d7 %in% knockdown_genes_zic1$gene]

genes_d7[genes_d7 %in% knockdown_genes_zic2$gene]

```

```{r}
library(UpSetR)
upset(fromList(list("D1" = genes_d1,
                    "D7" = genes_d7,
                    "all_peaks" = genes_all,
                    "KD_Zic1_decrease" = knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0],
                    "KD_Zic1_increase" = knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`>0],
                    "KD_Zic2_decrease" = knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0],
                    "KD_Zic2_increase" = knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`>0])),nsets = 7)


```
```{r}
upset(fromList(list("D1" = genes_d1,
                    "D7" = genes_d7,
                    "KD_Zic1_decrease" = knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0],
                    "KD_Zic1_increase" = knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`>0],
                    "KD_Zic2_decrease" = knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0],
                    "KD_Zic2_increase" = knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`>0])),nsets = 7)

```


```{r}
upset(fromList(list("D7" = genes_d7,
                    "KD_Zic1_decrease" = knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0],
                    "KD_Zic2_decrease" = knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0])),nsets = 7)

```

```{r}
upset(fromList(list("D1" = genes_d1,
                    "D7" = genes_d7,
                    "KD_Zic_decrease" = c(knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0],knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0]),
                    "KD_Zic_increase" = c(knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`>0],knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`>0]))))

```
# Putative direct targets
```{r}
genes_d7[genes_d7 %in% knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0]]


genes_d7[genes_d7 %in% knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0]]


```


# Plot overlaps from the genes perspectives

For each gene we want to know:
- Gene kd up or down
- Distance of tss to nearest Zic peak
- Is Zic peak differential up / down / none
- Is peak in vivo?

```{r}
library(tidyverse)
knockdown_genes_zic1_anno <- peaks_anno_list$all@anno[peaks_anno_list$all@anno$SYMBOL %in% c(knockdown_genes_zic1_up,knockdown_genes_zic1_down) ]


knockdown_genes_zic1_anno <- knockdown_genes_zic1_anno %>%
  as_tibble() %>%
  group_by(SYMBOL) %>%
  mutate(is_min = distanceToTSS == min(distanceToTSS)) %>%
  filter(is_min) %>%
  ungroup()


```

```{r}
options(scipen = 9999)
knockdown_genes_zic1_anno <- knockdown_genes_zic1_anno %>%
  mutate(SYMBOL = fct_reorder(SYMBOL, abs(distanceToTSS),.desc = TRUE),
         direction = case_when(
           log2FoldChange > 0  & padj < 0.05 ~ "up" ,
           log2FoldChange < 0  & padj < 0.05 ~ "down",
           log2FoldChange > 0  & padj > 0.05 ~ "up_ns" ,
           log2FoldChange < 0  & padj > 0.05 ~ "down_ns",
         ),
         padj = replace_na(padj, 1),
         knockdown_zic1 = case_when(
           SYMBOL %in% knockdown_genes_zic1_up ~ "up",
           SYMBOL %in% knockdown_genes_zic1_down ~"down",
           TRUE ~ "ns"
         ))
```


# For each knockdown gene, how close to its TSS is the nearest peak, and does it change?

```{r}
knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "down") %>%
ggplot(aes(  abs(distanceToTSS) + 2,SYMBOL, fill = direction)) +
  geom_col() +
  scale_x_log10() +
  geom_vline(xintercept = 6000) +
  scale_fill_manual(values = c("down" ="blue","down_ns"="gray","up"="red","up_ns"="gray"))

knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "up") %>%
ggplot(aes(  abs(distanceToTSS) + 2,SYMBOL, fill = direction)) +
  geom_col() +
  scale_x_log10() +
  geom_vline(xintercept = 6000) +
  scale_fill_manual(values = c("down" ="blue","down_ns"="gray","up"="red","up_ns"="gray"))

```
```{r}
knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "down") %>%
ggplot(aes(  abs(distanceToTSS) + 2,SYMBOL, fill = direction)) +
  geom_col() +
  scale_x_log10() +
  geom_vline(xintercept = 6000) +
  scale_fill_manual(values = c("down" ="blue","down_ns"="lightblue","up"="red","up_ns"="pink"))

knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "up") %>%
ggplot(aes(  abs(distanceToTSS) + 2,SYMBOL, fill = direction)) +
  geom_col() +
  scale_x_log10() +
  geom_vline(xintercept = 6000) +
  scale_fill_manual(values = c("down" ="blue","down_ns"="lightblue","up"="red","up_ns"="pink"))

```
# For each knockdown gene, how close to its TSS is the nearest significantly different peak, and what is the fold change?


# For each knockdown gene, how many peaks overlap, and are they significantly different?

read in vivo peaks
```{r}
peaks_7v1_invivo <- readRDS(file = "../data_output/diffbind_cutnrun_zic/peaks_7v1_invivo.Rds")

peaks_1v7_invivo <- readRDS(file = "../data_output/diffbind_cutnrun_zic/peaks_1v7_invivo.Rds")

peaks_7v1_invivo
```

```{r}
knockdown_genes_zic1_anno <- peaks_anno_list$all@anno[peaks_anno_list$all@anno$SYMBOL %in% c(knockdown_genes_zic1_up,knockdown_genes_zic1_down) ]
knockdown_genes_zic2_anno <- peaks_anno_list$all@anno[peaks_anno_list$all@anno$SYMBOL %in% c(knockdown_genes_zic2_up,knockdown_genes_zic2_down) ]

knockdown_genes_zic1_anno$invivo_up <- overlapsAny(knockdown_genes_zic1_anno,peaks_7v1_invivo,type = "equal")
knockdown_genes_zic1_anno$invivo_down <- overlapsAny(knockdown_genes_zic1_anno,peaks_1v7_invivo,type = "equal")

knockdown_genes_zic2_anno$invivo_up <- overlapsAny(knockdown_genes_zic2_anno,peaks_7v1_invivo,type = "equal")
knockdown_genes_zic2_anno$invivo_down <- overlapsAny(knockdown_genes_zic2_anno,peaks_1v7_invivo,type = "equal")


```

```{r}
library(tidyverse)
theme_set(theme_minimal())
theme_replace(aspect.ratio = 1)

knockdown_genes_zic1_anno <- knockdown_genes_zic1_anno %>%
  as_tibble() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL, abs(distanceToTSS),.desc = TRUE),
         direction = case_when(
           log2FoldChange > 0  & padj < 0.05 ~ "up" ,
           log2FoldChange < 0  & padj < 0.05 ~ "down",
           TRUE ~ "ns",
         ),
         padj = replace_na(padj, 1),
         knockdown_zic1 = case_when(
           SYMBOL %in% knockdown_genes_zic1_up ~ "up",
           SYMBOL %in% knockdown_genes_zic1_down ~"down",
           TRUE ~ "ns"
         ))

knockdown_genes_zic2_anno <- knockdown_genes_zic2_anno %>%
  as_tibble() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL, abs(distanceToTSS),.desc = TRUE),
         direction = case_when(
           log2FoldChange > 0  & padj < 0.05 ~ "up" ,
           log2FoldChange < 0  & padj < 0.05 ~ "down",
           TRUE ~ "ns",
         ),
         padj = replace_na(padj, 1),
         knockdown_zic2 = case_when(
           SYMBOL %in% knockdown_genes_zic2_up ~ "up",
           SYMBOL %in% knockdown_genes_zic2_down ~"down",
           TRUE ~ "ns"
         ))

```


Whole gene Zic1
```{r}
knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "down") %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic1 KD Down")

knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "up") %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic1 KD Up")

```

Whole gene Zic1 overlap in vivo
```{r}
knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "down",invivo_up | invivo_down) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic1 KD Down, in vivo overlap")

knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "up",invivo_up | invivo_down) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic1 KD Up, in vivo overlap")

```


whole gene Zic2 
```{r}
knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "down") %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic2 KD Down")

knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "up") %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic2 KD Up")

```
whole gene zic2 in vivo overlap
```{r}
knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "down", invivo_up | invivo_down) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic2 KD Down, in vivo overlap")

knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "up",invivo_up | invivo_down) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic2 KD Up, in vivo overlap")

```

Promoter
```{r}
knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "down",
         str_detect(annotation,"Promoter") ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic1 KD Up, Peaks at Promoters (+/- 3Kb TSS)")

knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "up",
          str_detect(annotation,"Promoter") ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic1 KD Down, Peaks at Promoters (+/- 3Kb TSS)")

```
```{r}
knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "down",
         str_detect(annotation,"Promoter") , invivo_up | invivo_down ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic1 KD Up, Peaks at Promoters (+/- 3Kb TSS), in vivo overlaps")

knockdown_genes_zic1_anno %>%
  filter(knockdown_zic1 == "up",
          str_detect(annotation,"Promoter") , invivo_up | invivo_down ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic1 KD Down, Peaks at Promoters (+/- 3Kb TSS), in vivo overlaps")

```


```{r}
knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "down",
         str_detect(annotation,"Promoter") ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic2 KD Up, Peaks at Promoters (+/- 3Kb TSS)")

knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "up",
          str_detect(annotation,"Promoter") ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic2 KD Up, Peaks at Promoters (+/- 3Kb TSS)")

```
```{r}
knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "down",
         str_detect(annotation,"Promoter"), invivo_up | invivo_down ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red"))+
  ggtitle("Zic2 KD Down, Peaks at Promoters (+/- 3Kb TSS), in vivo overlaps")

knockdown_genes_zic2_anno %>%
  filter(knockdown_zic2 == "up",
          str_detect(annotation,"Promoter"), invivo_up | invivo_down ) %>%
  group_by(SYMBOL, direction) %>%
  summarise(count = n()) %>%
  group_by(SYMBOL) %>%
  mutate(sum_count = sum(count)) %>%
  ungroup() %>%
  mutate(SYMBOL = fct_reorder(SYMBOL,sum_count)) %>%
  ggplot(aes(y = SYMBOL, x = count, fill = direction)) +
  geom_col() +
  scale_fill_manual(values = c("down" ="blue","ns"="darkgray","up"="red")) +
  ggtitle("Zic2 KD Up, Peaks at Promoters (+/- 3Kb TSS), in vivo overlaps")

```

# Distribution of fold changes

```{r}
knockdown_genes_zic1_anno %>%
  filter(str_detect(annotation,"Promoter")) %>%
  ggplot(aes(x = knockdown_zic1, y = log2FoldChange)) +
  geom_violin() +
  
  geom_jitter(data=knockdown_genes_zic1_anno %>%
                    filter(str_detect(annotation,"Promoter")),
              aes(x = knockdown_zic1, y = log2FoldChange, color = padj < 0.05),
              width = 0.2) +
    scale_color_manual(values=c("TRUE"="#3F3F3F","FALSE"="gray")) +
  geom_boxplot(outlier.shape = NA, width = 0.2)


knockdown_genes_zic2_anno %>%
  filter(str_detect(annotation,"Promoter")) %>%
  ggplot(aes(x = knockdown_zic2, y = log2FoldChange)) +
  geom_violin() +
  
  geom_jitter(data=knockdown_genes_zic2_anno %>%
                    filter(str_detect(annotation,"Promoter")),
              aes(x = knockdown_zic2, y = log2FoldChange, color = padj < 0.05),
              width = 0.2) +
    scale_color_manual(values=c("TRUE"="#3F3F3F","FALSE"="gray")) +
  geom_boxplot(outlier.shape = NA, width = 0.2)


```
```{r}
wilcox.test(knockdown_genes_zic1_anno$log2FoldChange[knockdown_genes_zic1_anno$knockdown_zic1 == "up"],
           knockdown_genes_zic1_anno$log2FoldChange[knockdown_genes_zic1_anno$knockdown_zic1 == "down"] )


wilcox.test(knockdown_genes_zic2_anno$log2FoldChange[knockdown_genes_zic2_anno$knockdown_zic2 == "up"],
           knockdown_genes_zic2_anno$log2FoldChange[knockdown_genes_zic2_anno$knockdown_zic2 == "down"] )

```


```{r}
knockdown_genes_zic1_anno %>%
  filter(str_detect(annotation,"Promoter"),
         padj < 0.05) %>%
  ggplot(aes(x = knockdown_zic1, y = log2FoldChange)) +
  geom_violin() +
  geom_jitter(width = 0.2) +
  geom_boxplot(outlier.shape = NA, width = 0.2)


knockdown_genes_zic2_anno %>%
  filter(str_detect(annotation,"Promoter"),
         padj < 0.05) %>%
  ggplot(aes(x = knockdown_zic2, y = log2FoldChange)) +
  geom_violin() +
  geom_jitter(width = 0.2, alpha = 0.75) +
  geom_boxplot(outlier.shape = NA, width = 0.2)

wilcox.test(knockdown_genes_zic1_anno$log2FoldChange[knockdown_genes_zic1_anno$knockdown_zic1 == "up"],
           knockdown_genes_zic1_anno$log2FoldChange[knockdown_genes_zic1_anno$knockdown_zic1 == "down"] )

```

```{r}
knockdown_genes_zic1_anno %>%
  filter(str_detect(annotation,"Promoter")) %>%
  group_by(SYMBOL,knockdown_zic1) %>%
  summarise(log2FoldChange = mean(log2FoldChange)) %>%
  ggplot(aes(x = knockdown_zic1, y = log2FoldChange)) +
  geom_violin() +
  geom_jitter(width = 0.2) +
  geom_boxplot(outlier.shape = NA, width = 0.2)


knockdown_genes_zic2_anno %>%
  filter(str_detect(annotation,"Promoter")) %>%
  group_by(SYMBOL,knockdown_zic2) %>%
  summarise(log2FoldChange = mean(log2FoldChange)) %>%
  ggplot(aes(x = knockdown_zic2, y = log2FoldChange)) +
  geom_violin() +
  geom_jitter(width = 0.2) +
  geom_boxplot(outlier.shape = NA, width = 0.2)

```

Notes:
- The ras gene was interesting because it has multiple peaks over its gene body that change, but not overlaps in vivo peaks, here is a paper that says ras pathway activated in activity dependent manner and requiredd for CGN survival in vitro https://www.sciencedirect.com/science/article/pii/S0021925820523442

- It does look like there is a tendency for going up in fold change for all differential genes. I think the peaks per promoter region and the violin plots show this.
- Doing the RNA version might be overkill if we already sort of know the answer? But couldn't hurt I guess.
- Also, I guess I should be doing the comparisons at DIV7 - DIV3



