---
title: "kdrnaseq_step4_rrho"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# RRHO of Zic KD with expression re-analyzed from Frank et al vs Tables from Chd7 cKO

Rank rank hypergeometric overlap
Adapted from https://github.com/WestLabDuke/Psychostimulant-NAcInterneuron/blob/main/scripts/RRHO.R

```{r}
library(tidyverse)
library(RRHO)

```

Read differentially expressed genes in Zic knockdowns

```{r}
knockdown_genes_zic1 <- read_tsv("../data_output/rnaseq_zic/diffexpr/res_zic1_ashr.tsv")
knockdown_genes_zic2 <- read_tsv("../data_output/rnaseq_zic/diffexpr/res_zic2_ashr.tsv")

```
Read differentially expressed genes in chd7 knockdowns

```{r}
knockdown_genes_chd7 <- readxl::read_excel("../data_input/rnaseq/GSE164360_RNA-seq_Summary.xlsx")
```




Change column names before the join

```{r}
colnames(knockdown_genes_zic1) <- paste0("zic1_",colnames(knockdown_genes_zic1))
colnames(knockdown_genes_zic2) <- paste0("zic2_",colnames(knockdown_genes_zic2))
colnames(knockdown_genes_chd7) <- paste0("chd7_",colnames(knockdown_genes_chd7))

```


```{r}
knockdown_genes_zic1_chd7 <- inner_join(knockdown_genes_zic1,
                                       knockdown_genes_chd7, 
                                       by = c("zic1_gene_names" = "chd7_GeneID"), 
                                       na_matches = "never")
knockdown_genes_zic1_chd7 <- knockdown_genes_zic1_chd7[!duplicated(knockdown_genes_zic1_chd7$zic1_gene_names) | duplicated(knockdown_genes_zic1_chd7$zic1_gene_names,fromLast = TRUE),]

knockdown_genes_zic2_chd7 <- inner_join(knockdown_genes_zic2,
                                       knockdown_genes_chd7, 
                                       by = c("zic2_gene_names" = "chd7_GeneID"),
                                       na_matches = "never")
knockdown_genes_zic2_chd7 <- knockdown_genes_zic2_chd7[!duplicated(knockdown_genes_zic2_chd7$zic2_gene_names) | duplicated(knockdown_genes_zic2_chd7$zic2_gene_names,fromLast = TRUE),]

```


# Prepare data for comparisons

```{r}
signal_zic1_vs_chd7 <- knockdown_genes_zic1_chd7 %>% 
  select(contains("adj"),contains("log2"),zic1_gene_names) %>%
  distinct() %>%
  drop_na() %>%
  mutate(sgnlogpadj = log10(zic1_padj)* sign(as.numeric(zic1_log2FoldChange))) %>% 
  dplyr::rename( SYMBOL = zic1_gene_names) %>% 
  dplyr::select(SYMBOL, sgnlogpadj) %>% 
  drop_na()

signal_chd7_vs_zic1 <- knockdown_genes_zic1_chd7 %>% 
  select(contains("adj"),contains("log2"),zic1_gene_names) %>%
  distinct() %>%
  drop_na() %>%
  mutate(sgnlogpadj = log10(`chd7_adj. P-Value`)* sign(as.numeric(chd7_log2FC))) %>% 
  dplyr::rename( SYMBOL = zic1_gene_names) %>% 
  dplyr::select(SYMBOL, sgnlogpadj) %>% 
  drop_na()

signal_chd7_vs_zic1_neg <- knockdown_genes_zic1_chd7 %>% 
  select(contains("adj"),contains("log2"),zic1_gene_names) %>%
  distinct() %>%
  drop_na() %>%
  mutate(sgnlogpadj = - log10(`chd7_adj. P-Value`)* sign(as.numeric(chd7_log2FC))) %>% 
  dplyr::rename( SYMBOL = zic1_gene_names) %>% 
  dplyr::select(SYMBOL, sgnlogpadj) %>% 
  drop_na()


signal_zic2_vs_chd7 <- knockdown_genes_zic2_chd7 %>% 
  select(contains("adj"),contains("log2"),zic2_gene_names) %>%
  distinct() %>%
  drop_na() %>%
  mutate(sgnlogpadj = log10(zic2_padj)* sign(as.numeric(zic2_log2FoldChange))) %>% 
  dplyr::rename( SYMBOL = zic2_gene_names) %>% 
  dplyr::select(SYMBOL, sgnlogpadj) %>% 
  drop_na()

signal_chd7_vs_zic2 <- knockdown_genes_zic2_chd7 %>% 
  select(contains("adj"),contains("log2"),zic2_gene_names) %>%
  distinct() %>%
  drop_na() %>%
  mutate(sgnlogpadj = log10(`chd7_adj. P-Value`)* sign(as.numeric(chd7_log2FC))) %>% 
  dplyr::rename( SYMBOL = zic2_gene_names) %>% 
  dplyr::select(SYMBOL, sgnlogpadj) %>% 
  drop_na()

signal_chd7_vs_zic2_neg <- knockdown_genes_zic2_chd7 %>% 
  select(contains("adj"),contains("log2"),zic2_gene_names) %>%
  distinct() %>%
  drop_na() %>%
  mutate(sgnlogpadj = - log10(`chd7_adj. P-Value`)* sign(as.numeric(chd7_log2FC))) %>% 
  dplyr::rename( SYMBOL = zic2_gene_names) %>% 
  dplyr::select(SYMBOL, sgnlogpadj) %>% 
  drop_na()

```

# RRHO

```{r}



object1 <- RRHO(as.data.frame(signal_zic1_vs_chd7),
                as.data.frame(signal_chd7_vs_zic1),
                BY=TRUE,
                stepsize = 100,
                alternative='two.sided',
                log10.ind = TRUE,
                plot = TRUE,
                outputdir = "../data_output/rnaseq_zic/rrho/",
                labels = c("Zic1_KD_Ctrl","Chd7_cKO_Ctrl"))

object2 <- RRHO(as.data.frame(signal_zic2_vs_chd7),
                as.data.frame(signal_chd7_vs_zic2),
                BY=TRUE,
                stepsize = 100,
                alternative='two.sided',
                log10.ind = TRUE,
                plot = TRUE,
                outputdir = "../data_output/rnaseq_zic/rrho/",
                labels = c("Zic2_KD_Ctrl","Chd7_cKO_Ctrl"))

object3 <- RRHO(as.data.frame(signal_zic1_vs_chd7),
                as.data.frame(signal_chd7_vs_zic1_neg),
                BY=TRUE,
                stepsize = 100,
                alternative='two.sided',
                log10.ind = TRUE,
                plot = TRUE,
                outputdir = "../data_output/rnaseq_zic/rrho/",
                labels = c("Zic1_KD_Ctrl","Chd7_Ctrl_cKO"))

object4 <- RRHO(as.data.frame(signal_zic2_vs_chd7),
                as.data.frame(signal_chd7_vs_zic2_neg),
                BY=TRUE,
                stepsize = 100,
                alternative='two.sided',
                log10.ind = TRUE,
                plot = TRUE,
                outputdir = "../data_output/rnaseq_zic/rrho/",
                labels = c("Zic2_KD_Ctrl","Chd7_Ctrl_cKO"))

```


