---
title: "zic_cutnrun_deseq"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Differential binding analysis of Zic CUT&RUN data

Steps:
- Get union of peaks across replicates / timepoints
- Get counts matrix
- Test

# Get union of peaks

## load cutnrun peaks as granges objects
```{r}
library(rtracklayer)
library(tidyverse)

cutnrun_peaks_dir <- "../data_output/cutnrun_zic/seacr_peaks/"
cutnrun_peaks_files <- list.files(cutnrun_peaks_dir,pattern = ".bed")
cutnrun_peaks_full <- paste0(cutnrun_peaks_dir,cutnrun_peaks_files)


cutnrun_peaks_granges <- lapply(cutnrun_peaks_full, import.bedGraph)

names(cutnrun_peaks_granges) <- str_split(cutnrun_peaks_files,"_",n=2) %>% map_chr(1)

names(cutnrun_peaks_granges) <- paste0("cutnrun-",names(cutnrun_peaks_granges))

```

## get union

```{r}
cutnrun_peaks_union_granges <- GenomicRanges::union(unlist(GRangesList(cutnrun_peaks_granges)),
                                                    unlist(GRangesList(cutnrun_peaks_granges)))



```

## remove peaks that overlap blacklist

```{r}
mm10_blacklist <- import("../data_input/bed/mm10-blacklist.v2.bed.gz")
cutnrun_peaks_union_granges <- cutnrun_peaks_union_granges[!overlapsAny(cutnrun_peaks_union_granges, mm10_blacklist)]

```



```{r}
elementMetadata(distanceToNearest(cutnrun_peaks_union_granges))$distance %>%
  summary()

```

```{r}
summary(width(cutnrun_peaks_union_granges))
```

```{r}
sum(width(cutnrun_peaks_union_granges) > 6000)
```

```{r eval = FALSE}
export.bed(cutnrun_peaks_union_granges,"../data_output/bed/cutnrun_zic_seacr_peaks_union.bed")
```


# Get counts matrix
```{r}
library(csaw)
bam_files <- list.files("../data_output/cutnrun_zic/bam/",pattern = ".bam$",full.names = T)
counts <- csaw::regionCounts(bam.files = bam_files,
                   regions = cutnrun_peaks_union_granges,
                   param = csaw::readParam(dedup = FALSE, minq = 255, pe = "both"))

colnames(counts) <- str_split(basename(bam_files),"_",2) %>% map_chr(1)

colData(counts)$rep <- str_split(colnames(counts),"-") %>% map_chr(1)
colData(counts)$time <- str_split(colnames(counts),"-") %>% map_chr(2)

```

Plot of total counts per sample
```{r}

options(scipen = 9999)
counts@colData %>% as_tibble() %>%
  ggplot(aes(totals/1e6, paste(time, rep))) +
  geom_col() +
  theme_minimal() +
  xlab("Total counts (Millions)") +
  ylab(NULL)

```

## Get DESeq object

```{r}
library(DESeq2)

deseq_obj <- DESeqDataSet(counts, design = ~ time)


```

# Prefiltering
```{r}
keep <- rowSums(counts(deseq_obj)) >= 20
deseq_obj <- deseq_obj[keep,]

```


# Test

Test with contrasts by time pairs from 1 to 3, 3 to 5, 5 to 7
```{r}
deseq_obj_contrasts <- DESeq(deseq_obj)

res_3v1 <- results(deseq_obj_contrasts,contrast = c("time","D3Z","D1Z"))
res_5v3 <- results(deseq_obj_contrasts,contrast = c("time","D5Z","D3Z"))
res_7v5 <- results(deseq_obj_contrasts,contrast = c("time","D7Z","D5Z"))

res_7v1 <- results(deseq_obj_contrasts,
                   contrast = c("time","D7Z","D1Z"),
                   alpha = 0.05,
                   altHypothesis = "greaterAbs")

```

Test with LRT analogous to anova (https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#likelihood-ratio-test)

```{r}
deseq_obj_lrt <- DESeq(deseq_obj, test="LRT", reduced=~1)
res_all <- results(deseq_obj_lrt)
```


P value histogram

```{r}
res_3v1 %>%
  as_tibble() %>%
  ggplot(aes(pvalue)) +
  geom_histogram()

res_5v3 %>%
  as_tibble() %>%
  ggplot(aes(pvalue)) +
  geom_histogram()

res_7v5 %>%
  as_tibble() %>%
  ggplot(aes(pvalue)) +
  geom_histogram()


res_7v1 %>%
  as_tibble() %>%
  ggplot(aes(pvalue)) +
  geom_histogram()

res_all %>%
  as_tibble() %>%
  ggplot(aes(pvalue)) +
  geom_histogram()

```



# Shrinkage
```{r}
resultsNames(deseq_obj_contrasts)

res_7v1_ashr <- lfcShrink(deseq_obj_contrasts,
                          res  = res_7v1,
                          type="ashr")

```

```{r}
plotMA(res_7v1_ashr, ylim = c(-3,3))

```

```{r}
rowRanges(deseq_obj)[res_7v1_ashr$log2FoldChange > 2.2]
```


```{r}
res_7v1_ashr %>%
  as_tibble() %>%
  ggplot(aes(log2FoldChange, -10 * log10(padj))) +
  geom_point(aes(color = padj < 0.05 & abs(log2FoldChange) > 1)) +
  scale_color_manual(values = c("FALSE" = "gray","TRUE"="blue")) +
  theme_classic() +
  theme(aspect.ratio = 1) 
```
# Plots

## Volcano / MA
```{r}
options(scipen = 99999)

plotMA(res_3v1, alpha = 0.05) 
plotMA(res_5v3, alpha = 0.05)
plotMA(res_7v5, alpha = 0.05)
plotMA(res_7v1, alpha = 0.05, ylim = c(-6,6))

res_7v1_noshrink <- res_7v1
res_7v1 <- res_7v1_ashr
```

```{r}
sum(replace_na(res_7v1$padj,1) < 0.05 & res_7v1$log2FoldChange > 1 )
sum(replace_na(res_7v1$padj,1) < 0.05 & res_7v1$log2FoldChange < -1 )

```


## Heatmap
Get vst counts
```{r}
vsd <- vst(deseq_obj, blind=FALSE)


```

```{r}
library(ComplexHeatmap)


plot_diffbind_heatmap <- function(vst_tab, res_obj){
  scaled_vst <- t(scale(t(assay(vst_tab[tidyr::replace_na(res_obj$padj,1) < 0.05,]))))
  colnames(scaled_vst) <- paste((str_split(colnames(scaled_vst),"-") %>% map_chr(2)),
                                (str_split(colnames(scaled_vst),"-") %>% map_chr(1)),sep = "-")
  
  scaled_vst <- scaled_vst[, order(colnames(scaled_vst))]
  
  Heatmap(scaled_vst ,
          cluster_columns = FALSE,
          show_row_dend = FALSE,
          col = viridis::inferno(n = 2,begin = 0.2, end = 0.8))
  
} 




```

```{r}
plot_diffbind_heatmap(vsd, res_3v1) + ggtitle("DIV3-DIV1")
plot_diffbind_heatmap(vsd, res_5v3)+ ggtitle("DIV5-DIV3")
plot_diffbind_heatmap(vsd, res_7v5)+ ggtitle("DIV7-DIV5")
plot_diffbind_heatmap(vsd, res_7v1) + ggtitle("DIV7-DIV1")
plot_diffbind_heatmap(vsd, res_all)+ ggtitle("Any timepoint")
```


```{r}
  scaled_vst <- t(scale(t(assay(vsd[tidyr::replace_na(res_7v1$padj,1) < 0.05,]))))
  colnames(scaled_vst) <- paste((str_split(colnames(scaled_vst),"-") %>% map_chr(2)),
                                (str_split(colnames(scaled_vst),"-") %>% map_chr(1)),sep = "-")
  
  scaled_vst <- scaled_vst[, order(colnames(scaled_vst))]
  
  Heatmap(scaled_vst ,
          cluster_columns = FALSE,
          show_row_dend = FALSE,
          row_km = 2,
          col = viridis::inferno(n = 2,begin = 0.2, end = 0.8))

```
```{r}
  Heatmap(scaled_vst ,
          cluster_columns = FALSE,
          show_row_dend = TRUE,
          col = viridis::inferno(n = 2,begin = 0.2, end = 0.8))

```

## Aluviall 


Notes: 

- In general there seems to be three large groups:

- Peaks that loose binding 1 -> 7
- Peaks that gain binding 1 -> 7 
For both of these there might be subgroups of different gain/loss timing (i.e. some binding is gained by day 3, others 5 etc)
- Peaks gain and then loose (gain by 3 then start loosing) Could it be due to cell culture?



## Annotation: Assign peaks to nearest gene  
```{r}
library(ChIPseeker)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
txdb <- TxDb.Mmusculus.UCSC.mm10.knownGene

peaks_7v1 <- rowRanges(deseq_obj_contrasts[tidyr::replace_na(res_7v1$padj,1) < 0.05 &
                                             tidyr::replace_na(res_7v1$log2FoldChange,0) > 0])
peaks_1v7 <- rowRanges(deseq_obj_contrasts[tidyr::replace_na(res_7v1$padj,1) < 0.05 &
                                             tidyr::replace_na(res_7v1$log2FoldChange,0) < 0])


peaks_7v1_anno <- annotatePeak(peak = peaks_7v1,TxDb = txdb,
                               level = "gene",
                               assignGenomicAnnotation = T,
                               annoDb = "org.Mm.eg.db")
peaks_1v7_anno <- annotatePeak(peak = peaks_1v7,
                               TxDb = txdb,
                               level = "gene",
                               assignGenomicAnnotation = T,
                               annoDb = "org.Mm.eg.db")
peaks_all_anno <- annotatePeak(peak = rowRanges(deseq_obj_contrasts),
                               TxDb = txdb,
                               level = "gene",
                               assignGenomicAnnotation = T,
                               annoDb = "org.Mm.eg.db")

```



```{r}

peaks_anno_list <- list("D7" = peaks_7v1_anno,
                 "D1" = peaks_1v7_anno,
                 "all" = peaks_all_anno)

plotAnnoBar(peaks_anno_list)

```


# Overlap with in vivo peaks
```{r}
get_overlaps_by_element <- function(peaks_a, peaks_b) {
                              intersection1 <- sum(IRanges::overlapsAny(peaks_a,peaks_b))
                              setdiff1 <- length(peaks_a) - intersection1
                              intersection2 <- sum(IRanges::overlapsAny(peaks_b,peaks_a))
                              setdiff2 <-length(peaks_b) - intersection2
                              
                              return(data.frame("intersection1" = intersection1,
                                                "intersection2" = intersection2,
                                                "setdiff1" = setdiff1,
                                                "setdiff2" = setdiff2))}

```

## load ChIP peaks - reprocessed seacr
```{r}
chip_seacr_peaks_dir <- "../data_output/chip_zic/chip_zic_seacr_peaks/"
chip_seacr_peaks_files <- list.files(chip_seacr_peaks_dir,pattern = "bed$")
chip_seacr_peaks_files <- chip_seacr_peaks_files[1:4]
chip_seacr_peaks_full <- paste0(chip_seacr_peaks_dir,chip_seacr_peaks_files)
chip_seacr_peaks_names <- c("P7","P7","P60","P60")
chip_seacr_peaks_names <- paste0("ChIP-seacr-","rep",rep(1:2,2),"-",chip_seacr_peaks_names)

chip_seacr_granges <- lapply(chip_seacr_peaks_full, import.bedGraph)

names(chip_seacr_granges) <- chip_seacr_peaks_names

```

## load ChIP peaks - reprocessed Melyssa
```{r}
chip_macs_peaks_dir <- "../data_input/melyssa_zic_data/"
chip_macs_peaks_files <- list.files(chip_macs_peaks_dir,pattern = "bed$")
chip_macs_peaks_full <- paste0(chip_macs_peaks_dir,chip_macs_peaks_files)
chip_macs_peaks_names <- c("P7","NS","P60")

chip_macs_granges <- lapply(chip_macs_peaks_full, import.bedGraph)

names(chip_macs_granges) <- chip_macs_peaks_names

```




```{r}

chip_seacr_granges_by_time <- split(chip_seacr_granges, str_split(names(chip_seacr_granges),"-",4) %>% map_chr(4))



chip_seacr_granges_reduced <- lapply(chip_seacr_granges_by_time, function(x) {
  Reduce(GenomicRanges::intersect, x)
})

```

```{r}
all_peaks_overlaps_by_element <- rbind(get_overlaps_by_element(peaks_1v7, chip_seacr_granges_reduced$P7),
get_overlaps_by_element(peaks_1v7, chip_seacr_granges_reduced$P60),
get_overlaps_by_element(peaks_7v1, chip_seacr_granges_reduced$P7),
get_overlaps_by_element(peaks_7v1, chip_seacr_granges_reduced$P60))


all_peaks_overlaps_by_element$set1 <- c("peaks_1v7" , "peaks_1v7", "peaks_7v1", "peaks_7v1")
all_peaks_overlaps_by_element$set2 <- c("P7", "P60", "P7", "P60")
```


```{r}
diff_peaks_overlaps_by_element <- rbind(get_overlaps_by_element(peaks_1v7, chip_macs_granges$P7),
get_overlaps_by_element(peaks_1v7, chip_macs_granges$P60),
get_overlaps_by_element(peaks_7v1, chip_macs_granges$P7),
get_overlaps_by_element(peaks_7v1, chip_macs_granges$P60))


diff_peaks_overlaps_by_element$set1 <- c("peaks_1v7" , "peaks_1v7", "peaks_7v1", "peaks_7v1")
diff_peaks_overlaps_by_element$set2 <- c("P7", "P60", "P7", "P60")

```





```{r}
all_peaks_overlaps_by_element %>%
  dplyr::select(-setdiff2, -intersection2) %>%
  pivot_longer(is.numeric) %>%
    group_by(set1,set2) %>%
    mutate(percentage = 100 * value / sum(value)) %>%
    ungroup() %>%
    mutate(name = fct_relevel(name,"setdiff1","intersection1"),
           name = fct_recode(name, "overlap" = "intersection1", "no overlap" = "setdiff1"),
           set2 = fct_relevel(set2,"P7","P60")) %>%
    ggplot(aes(y = percentage, x = set2, fill = name)) +
    geom_col(width = 0.5) +
    ylab(NULL) +
    ylab("Percentage") +
  xlab(NULL) +
    scale_fill_viridis_d(begin = 0.2,end = 0.8) +
    theme_light() +
  theme(aspect.ratio = 1.5,legend.title = element_blank()) +
  facet_grid(~set1)
```


```{r}
diff_peaks_overlaps_by_element %>%
  dplyr::select(-setdiff2, -intersection2) %>%
  pivot_longer(is.numeric) %>%
    group_by(set1,set2) %>%
    mutate(percentage = 100 * value / sum(value)) %>%
    ungroup() %>%
    mutate(name = fct_relevel(name,"setdiff1","intersection1"),
           name = fct_recode(name, "overlap" = "intersection1", "no overlap" = "setdiff1"),
           set2 = fct_relevel(set2,"P7","P60")) %>%
    ggplot(aes(y = percentage, x = set2, fill = name)) +
    geom_col(width = 0.5) +
    ylab(NULL) +
    ylab("Percentage") +
  xlab(NULL) +
    scale_fill_viridis_d(begin = 0.2,end = 0.8) +
    theme_light() +
  theme(aspect.ratio = 1.5,legend.title = element_blank()) +
  facet_grid(~set1)
```

Get list of peaks that overlap

```{r}
peaks_1v7_invivo <- peaks_1v7[overlapsAny(peaks_1v7, chip_macs_granges$P7)]
peaks_7v1_invivo <- peaks_7v1[overlapsAny(peaks_7v1, chip_macs_granges$P60)]

```

```{r}
peaks_1v7_invivo_anno <- annotatePeak(peak = peaks_1v7_invivo,TxDb = txdb,
                               level = "gene",
                               assignGenomicAnnotation = T,
                               annoDb = "org.Mm.eg.db")
peaks_7v1_invivo_anno <- annotatePeak(peak = peaks_7v1_invivo,
                               TxDb = txdb,
                               level = "gene",
                               assignGenomicAnnotation = T,
                               annoDb = "org.Mm.eg.db")

```

```{r}
peaks_anno_invivo_list <- list("D7" = peaks_7v1_invivo_anno,
                 "D1" = peaks_1v7_invivo_anno)

plotAnnoBar(peaks_anno_invivo_list)


```


# Gene set enrichment
```{r}
library(clusterProfiler)

gene_ids_list <- lapply(peaks_anno_list,
       function(x) x@anno$geneId)

comp_go <- compareCluster(geneCluster   = gene_ids_list,
                         fun           = "enrichGO",
                         pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",
                         OrgDb = org.Mm.eg.db)

dotplot(comp_go)

```

```{r}
gene_ids_list_invivo <- lapply(peaks_anno_invivo_list,
       function(x) x@anno$geneId)

comp_go <- compareCluster(geneCluster   = gene_ids_list_invivo,
                         fun           = "enrichGO",
                         pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",
                         OrgDb = org.Mm.eg.db)

dotplot(comp_go)

```


```{r}
gene_anno_list <- lapply(peaks_anno_list,
       function(x) x@anno[abs(x@anno$distanceToTSS) < 2000])

gene_anno_list_invivo <- lapply(peaks_anno_invivo_list,
       function(x) x@anno[abs(x@anno$distanceToTSS) < 2000])

```


```{r}
genes_d7 <- unique(gene_anno_list$D7$SYMBOL)
genes_d1 <- unique(gene_anno_list$D1$SYMBOL)

genes_d7_invivo <- unique(gene_anno_list_invivo$D7$SYMBOL)
genes_d1_invivo <- unique(gene_anno_list_invivo$D1$SYMBOL)

genes_all <- unique(peaks_anno_list$all@anno$SYMBOL)

length(genes_all)

unique(peaks_anno_list$all@anno$SYMBOL[peaks_anno_list$all@detailGenomicAnnotation$Promoter])
```

Check example genes
```{r}

any(str_detect(genes_d1,"Atoh1"))
any(str_detect(genes_d1,"Elavl3"))

any(str_detect(genes_d7,"Cbln3"))
any(str_detect(genes_d7,"Gabra6"))

```

## Do differentially bound genes overlap differentially knockdown genes?

Read differentially expressed genes

```{r}
knockdown_genes_zic1 <- readxl::read_excel("../data_input/melyssa_zic_data/41593_2015_BFnn3995_MOESM88_ESM.xls")
knockdown_genes_zic2 <- readxl::read_excel("../data_input/melyssa_zic_data/41593_2015_BFnn3995_MOESM88_ESM.xls",sheet = 2)

```

```{r}
sum(genes_d1 %in% knockdown_genes_zic1$gene)
sum(genes_d7 %in% knockdown_genes_zic1$gene)

sum(genes_d1 %in% knockdown_genes_zic2$gene)
sum(genes_d7 %in% knockdown_genes_zic2$gene)

genes_d1[genes_d1 %in% knockdown_genes_zic1$gene]
genes_d7[genes_d7 %in% knockdown_genes_zic1$gene]

genes_d1[genes_d1 %in% knockdown_genes_zic2$gene]
genes_d7[genes_d7 %in% knockdown_genes_zic2$gene]

```

```{r}
library(UpSetR)
upset(fromList(list("D1" = genes_d1,
                    "D7" = genes_d7,
                    "all_peaks" = genes_all,
                    "KD_Zic1_decrease" = knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0],
                    "KD_Zic1_increase" = knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`>0],
                    "KD_Zic2_decrease" = knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0],
                    "KD_Zic2_increase" = knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`>0])),nsets = 6)


```

```{r}
upset(fromList(list("D1" = genes_d1,
                    "D7" = genes_d7,
                    "KD_Zic_decrease" = c(knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0],knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0]),
                    "KD_Zic_increase" = c(knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`>0],knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`>0]))))

```
# Putative direct targets
```{r}
genes_d7[genes_d7 %in% knockdown_genes_zic1$gene[knockdown_genes_zic1$`log2(Fold-Change)`<0]]


genes_d7[genes_d7 %in% knockdown_genes_zic2$gene[knockdown_genes_zic2$`log2(Fold-Change)`<0]]


```


```{r}

upset(fromList(list("D1" = genes_d1_invivo,
                    "D7" = genes_d7_invivo,
                    "KD_Zic1" = knockdown_genes_zic1$gene,
                    "KD_Zic2" = knockdown_genes_zic2$gene)))

```

```{r}
genes_d1_invivo[genes_d1_invivo %in% knockdown_genes$gene]
genes_d7_invivo[genes_d7_invivo %in% knockdown_genes$gene]

```


Notes:
- Goal was to annotate peaks in terms of where they occur in the genome
  - Most peaks are distal
  - No change in distribution between early and late peaks
- Assigning peaks to nearest tss and look for functional enrichments (nearest means at most 6kb from nearest tss)
  - Note: using all genes (nearest gene but any gene part) does not give different results for each type of cluster
  - Some relevant genes do show changes: Atoh, arc, Elav3, Cbln3, Gabra6
  - Gain of Zic binding in GABA receptor GO category
  - The early one has some ion channel activity related GO


- Overlap with rnaseq
Only a few genes overlap. We could view these as "putative direct targets" but they don't look that great on the browser.
- Overlap with previous chip seq
There is a subset of peaks that gain binding in both in vivo and in culture samples.
- Shrinkage
Main takeaway is the most compelling sites that gain binding are not promoters. That is the peaks with highest logFC are not near promoters. Sometimes they are on gene bodies though.

Next steps:
- Do these gain / loss peaks are also gained / lost in vivo ?
- Plot IGV example regions of up and down
- Do shrinkage for sorting genes and for plots  

- Characterize other timepoints
- Try annotating "distal" peaks to the cerebellum enhancer - promoter peaks: hic files in data_input/bedpe
Hic files are from two papers where they do hic in P4 and P22.
p4
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE164360
p22
https://pubmed.ncbi.nlm.nih.gov/32647123/
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4120009
- Deeptools plots showing the gain / loss of peaks 

- Correlate with diff expressed genes ?
https://www.ncbi.nlm.nih.gov/gds/?term=Regulation%20of%20chromatin%20accessibility%20and%20Zic%20binding%20at%20enhancers%20in%20the%20developing%20cerebellum


- Another idea was to overlap to known enhancers
http://www.enhanceratlas.org/downloadv2.php
http://www.enhanceratlas.org/data/AllEPs/hs/Cerebellum_EP.txt
